### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by lathonez** [lathonez/clicker#38]
Feature Roadmap:

* linting
* testing with watches
* typescript maps for debug/coverage
* examples of tests mocking observables/promises
* put example code in module structure for easy removal and showing pattern
* build location for custom images used
* e2e testing with protractor
* release to app store(s) with travis
* map page
* graphs / charts page

**Comment by lathonez** [comment]
@marcmeans - FYI

**Comment by lathonez** [comment]
**Linting**

tslint should work out of the box if you have a tslint.json. If you would like further enhancements to linting please raise an issue and explain there.

**Comment by lathonez** [comment]
**Testing with watches**

Was implemented in #19. Explained in the blog and readme. If you would like further enhancements to this please raise an issue and explain there.

**Comment by lathonez** [comment]
**examples of tests mocking observables/promises**

I don't want this repo to turn into a "how to unit test" thing. It's a rabbit hole and I don't have the time or inclination to support it.

That being said there are certain things that I think are worth including, for example, testing Ionic @Pages using angular2 dependency injection, which is not done yet - though components are.

The best place for general unit testing questions probably stack overflow. There is an example with promises in [clickers.spec.ts](https://github.com/lathonez/clicker/blob/master/app/services/clickers.spec.ts#L75-L81) (most of the complex examples are therein).

**Comment by lathonez** [comment]
** put example code in module structure for easy removal and showing pattern **

@marcmeans - sounds interesting, but I don't know what you mean? Maybe raise an issue with some explanation?

**Comment by geoHeil** [comment]
What about Hot Module Replacement?

**Comment by lathonez** [comment]
@geoHeil are you suggesting a move back to webpack (wouldn't happen unless Ionic went back again), or using [browserify-hmr](https://github.com/AgentME/browserify-hmr)?

**Comment by geoHeil** [comment]
Not sure about implementation details, just saying HMR would be cool. As such `browserify-hmr` seems to be fine.

**Comment by lathonez** [comment]
> just saying HMR would be cool

agreed

**Comment by ddellamico** [comment]
In this [post](http://blog.ionic.io/announcing-ionic-2-beta-11/), seems Ionic team are migrating again from Browserify to Webpack 2 to take advantage of "tree shaking." .. It would be nice start moving this project to webpack.. what do you think ?

**Comment by lathonez** [comment]
Yeah, if they do end up going with webpack2 in beta12 we'll switch over as
part of incorporating that upgrade into this repo.

I don't think it's worth doing ahead of time.

On 13 Aug 2016 02:44, "Damien" <notifications@github.com> wrote:

In this post <http://blog.ionic.io/announcing-ionic-2-beta-11/>, seems
Ionic team are migrating again from Browserify to Webpack 2 to take
advantage of "tree shaking." .. It would be nice start moving this project
to webpack.. what do you think ?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/lathonez/clicker/issues/38#issuecomment-239497285>, or mute
the thread

**Comment by marcoturi** [comment]
Would be good to have same code comments, maybe through typedoc ?

**Comment by lathonez** [comment]
@marcoturi - agreed. I've opened #139.

If you're feeling generous, a small PR would be fantastic.

If not thanks a lot for the suggestion, will do my best get round to it!

**Comment by thielCole** [comment]
@lathonez Have you put any thought into doing e2e testing on actual devices or through an emulator?

**Comment by lathonez** [comment]
#52 (top of the list). Last time I looked SauceLabs was the best option for FOSS.

Realistically I don't have any time for moving this roadmap forward (until my associated closed source project requires it).

Very happy for PRs / discussion suggestions around the correct direction.

**Comment by thielCole** [comment]
Ah it was hidden in the name. Yeah I'm currently checking out both a SauceLabs solution and Appium.

**Comment by lathonez** [comment]
If you come up with something that works let us know, maybe share config.

Thanks

**Comment by lathonez** [comment]
Now we're using @angular/cli, anything in the test direction should be raised upstream there. We'll take any features they add.
